pyam/aggregate.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+import logging
+
+from pyam.logging import adjust_log_level
+from pyam.utils import (
+    islistable,
+    isstr,
+    KNOWN_FUNCS,
+    META_IDX,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def aggregate(df, variable, components=None, method=np.sum):
+    """Internal implementation of the `aggregate` function"""
```

Comment:
no, the public API should be accessed via `IamDataFrame.aggregate()`

Reply:
ah ok. In that case it's tricky because the way they're currently written (without the underscore) suggests that they are in the public API. In https://github.com/danielhuppmann/pyam/pull/14, I put everything in the private API and then use that in the core module. That goes against what @gidden suggested re not using things outside their modules but is something I've seen elsewhere (e.g. pandas plotting stuff https://github.com/pandas-dev/pandas/tree/master/pandas/plotting) so might a suitable compromise given they shouldn't be in the public API?

Reply:
thanks, merged your PR